.github/workflows/integration-tests.yml

```diff
@@ -102,24 +102,89 @@ jobs:
 
   psycopg2-check:
     name: "Test psycopg2 build version"
-    runs-on: ${{ matrix.scenario.platform }}
+    runs-on: ${{ matrix.platform }}
     strategy:
       fail-fast: false
       matrix:
-        scenario:
-          - {platform: ubuntu-latest, psycopg2-name: psycopg2}
-          - {platform: macos-12, psycopg2-name: psycopg2-binary}
+        platform: [ubuntu-22.04, macos-12]
+        python-version: ["3.8", "3.11"]
     steps:
       - name: "Check out repository"
         uses: actions/checkout@v4
 
-      - name: "Test psycopg2 name"
+      - name: "Set up Python ${{ matrix.python-version }}"
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: "Test psycopg2 name - default"
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+          python -m pip install .
+
+          PSYCOPG2_NAME=$((pip show psycopg2 || pip show psycopg2-binary) | grep Name | cut -d " " -f 2)
```

Comment:
A couple of full shell best practices.

1. with the inner `()` I'd recommend piping error channels out to `/dev/null`
2. I'd actually propose a different approach in general:
```shell
pip list --format=json | jq -r '.[] | select(.name == "psycopg2" or .name == "psycopg2-binary") | .name'
```
jq is lightweight and it allows you to more programmatically look for you want rather than relying on unstructured string output

Reply:
Will that work with `pip show` too? I was using `pip show` because it only returns that specific package and the existence of `psycopg2-binary` wouldn't be triggered when searching for `psycopg2`. When using `pip list` or `pip freeze`, I needed to look for `"psycopg2 "` to find only `psycopg2` and not `psycopg2-binary`.

Reply:
So it looks like `jq` is not installed by default. Given that the intent of this story is to avoid making users install os deps, I'd like to stick with the method that only relies on `pip`. Do you have any concerns with that approach? Is there a better way that doesn't use additional os dependencies that also doesn't use string parsing?